### PR TITLE
refactor: use parking lot in the Read Buffer

### PR DESCRIPTION
**Pull request by e-dard**
Not much to say, I've been meaning to do this for weeks.

### CODE REVIEW

File: read_buffer/Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ hashbrown = "0.9.1"
 internal_types = { path = "../internal_types" }
 itertools = "0.9.0"
 packers = { path = "../packers" }
+parking_lot = "0.11.1"
```
Comment (alamb) [inline read_buffer/Cargo.toml:22]:
```suggestion
parking_lot = "0.11"
```

File: server/src/db.rs [hunk 2]
```diff
@@ -1530,7 +1530,7 @@ mod tests {
                 to_arc("1970-01-01T00"),
                 0,
                 ChunkStorage::ReadBuffer,
-                1285,
+                1269,
```
Comment (alamb) [inline server/src/db.rs:1533]: this change just reflects the size of the parking lot mutex structure itself, right?

**Review by alamb — APPROVED** [review]
This looks much better to me

**Review by domodwyer — APPROVED** [review]
I'd be interested to know if this has any noticeable perf improvements? The docs say (and I've always wondered about):

> Inline fast path for the uncontended case.

File: server/src/db.rs (hunk 2 above)
Comment (e-dard) [reply server/src/db.rs:1533]: Yes, see https://github.com/influxdata/influxdb_iox/issues/1217

**Pull request merged**